cargo-guppy/src/core.rs

```diff
@@ -109,30 +112,33 @@ impl FilterOptions {
                 Kind::Workspace => from.in_workspace() && to.in_workspace(),
             };
 
-            // filter out irrelevant dependencies for a specific target (--target)
-            let include_target = if let Some(platform) = &platform {
-                edge.normal()
-                    .map(|meta| {
-                        // Include this dependency if it's optional or required or if the status is
-                        // unknown.
-                        meta.enabled_on(platform) != EnabledStatus::Never
-                    })
-                    .unwrap_or(true)
+            let include_type = if let Some(platform) = &platform {
+                // filter out irrelevant dependencies for a specific target (--target)
+                self.eval(edge, |req| {
+                    req.status().enabled_on(platform) != IsEnabled::Disabled
+                })
             } else {
-                true
+                // keep dependencies that are potentially enabled on any platform
+                self.eval(edge, |req| req.is_present())
             };
 
-            // filter normal, dev, and build dependencies (--include-build, --include-dev)
-            let include_type = edge.normal().is_some()
-                || self.include_dev && edge.dev().is_some()
-                || self.include_build && edge.build().is_some();
-
             // filter out provided edge targets (--omit-edges-into)
             let include_edge = !omitted_package_ids.contains(to.id());
 
-            include_kind && include_target && include_type && include_edge
+            include_kind && include_type && include_edge
         }
     }
+
+    /// Filter normal, dev, and build dependencies (--include-build, --include-dev).
+    fn eval(
+        &self,
+        edge: PackageEdge<'_>,
+        mut pred_fn: impl FnMut(DependencyReq<'_>) -> bool,
+    ) -> bool {
+        pred_fn(edge.normal())
+            || self.include_dev && pred_fn(edge.dev())
+            || self.include_build && pred_fn(edge.build())
+    }
```

Comment:
I do think this is potentially cleaner, but the comments don't quite match up to what is happening here and at the callsite. This function filters both by normal/dev/build and then also by a requirement predicate, but the predicate isn't mentioned in the comments here and the dev/build/normal filtering isn't mentioned at the call site. Can you improve the comments a bit?

Reply:
Will do.